lua/orgmode/parser/file.lua

```diff
@@ -204,7 +204,7 @@ end
 ---@return Section[]
 function File:find_headlines_by_title(title)
```

Comment:
I just figured out that this method is also used for autocompletion, which should actually do matching only by the beginning of the string. Let's add a second argument called `strict` which would default to true, and do matching as it is done now. If `strict` is false, do it the old way.
Once that's added, just pass `false` [here](https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/org/hyperlinks.lua#L70).
Another option, maybe even a bit cleaner, is to introduce a method called `File:find_headlines_matching_title` that would behave in the old way, and use that in hyperlinks autocompletion.

Reply:
I'd prefer the version where we pass a second argument, as I'd rather reduce than increase the amount of functions we have. Also `exact` would be better than `strict` in my opinion. Just let me know which you want to settle on.

Reply:
Sure that works. Exact is better than strict.

Reply:
What about here: https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/org/hyperlinks.lua#L106 ?

Reply:
There we don't need strict matching, just start needs to match.

Reply:
@kristijanhusak pushed some changes, should be working now